pretty_gpx/common/data/overpass_processing.py

```diff
@@ -3,6 +3,7 @@
 
```

Comment:
@ThomasParistech Here are some data with my internet connection (Cellphone hotspot) 
If you make additional overpass queries that are not in the large query where we download the data you increase significantly the time needed.

For example on the paris marathon full precision: 4 overpass requests are made:
- 2 with Nominatim for start and end name : 7.2s on my connection
- 1 large download for the whole data : 25s download
- 1 count download to get the number of ways in each category to get to download the correct precision for roads: 4.5s

In comparison to : 
- Simplify roads called 2k times which takes for the 2k calls 3.2s (1.2 for the merging and 2 for the shapely simplify).

So it depends what we want.

Keeping full precision but adding the simplify roads function reduce the size of the paris marathon poster down to 12MB. If you also use the count function to detect that you need to download only 3 roads types out of 4 then the size reduces down to 7MB.

Reply:
@ThomasParistech Seeing this data I split the PR in 2

Reply:
@JulesL2 
Sorry for the late reply

It seems like keeping full precision and simplifying is a good tradeoff

I really like your idea of adding a slider in the gui to select the roads to draw, since it means we don't have to design the road-type selection heuristic ourselves

Reply:
@ThomasParistech Then this PR is ready to be merged, it has only the simplify roads features

Reply:
:+1: 
I'll handle the mypy error with flyio